cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/Transform.java

```diff
@@ -61,7 +62,29 @@ public void destroy() {
     //no-op
   }
 
+  /**
+   * Prepare the Batch run. Used to configure the job before starting the run.
+   *
+   * @param context transform prepare context
+   * @throws Exception if there's an error during this method invocation
+   */
+  public void prepareRun(TransformPrepareContext context) throws Exception {
```

Comment:
it seems like these methods should be a part of StageLifecycle and supported for all plugin types, not just transforms, source, and sinks.